### ISSUE TITLE
[http-request] : Can't set payload handling when method is passed in via `msg.method`

### ISSUE THREAD

**Issue by marcus-j-davies** [node-red/node-red#3957]
### Current Behavior

When `- set by msg.method -` is used for the method, it hides the payload handling dropdown on the node config editor.
Now then, If the user does not first choose the payload handling option prior to  selecting `- set by msg.method -` 

The users payload is ignored, i.e for GET requests, the payload is not passed in as a query string, as the default is to ignore it.

The user has to first set it back to GET, change the payload handling option, then set it back to `- set by msg.method -`.
The option is there initially, but this could trip up new users.

### Expected Behavior

Iether allow payload handling options to be include in the message, or to not hide the option

### Steps To Reproduce

- Add a new http-request node
- set it to `- set by msg.method -`
- Note that the payload handling option has gone 

### Environment

- Node-RED version: **3.0.0**
- Node.js version: **16.15.1**
- npm version: **8.11.0**
- Platform/OS: **Debian**
- Browser: **Safari**

**Comment by Paul-Reed** [comment]
I have an example flow which I can share by forum DM if necessary.

**Comment by Steve-Mcl** [comment]
@Paul-Reed could you share a minimal flow here please?

Ta.

**Comment by Paul-Reed** [comment]
No, because it contains my api key/secret, but I can DM it to you, for your personal attention.
The error cannot be demonstrated without a valid api request which needs api auth.

**Comment by Steve-Mcl** [comment]
Issue confirmed.

> ether allow payload handling options to be include in the message, or to not hide the option

@dceejay my preference would be to permit the payload handling option (for GET requests) to be set by the `msg` rather than always display the dropdown. 

My reasoning is there may be only 1 HTTP Request Node on a flow, used for all GET/POST/ETC using `msg.method` and some of the routes may want payload in the QS, some may want payload in the body.

My current solution would be to replace the at lines 203~205
https://github.com/node-red/node-red/blob/master/packages/node_modules/@node-red/nodes/core/network/21-httprequest.js#L203-L205

... with a "per msg" (on the fly) determination...

```
        this.on("input",function(msg,nodeSend,nodeDone) {
            // ...
            // ... snipped for brevity ---
            // ...
            // ... 
            let paytoqs = false
            let paytobody = false
            if (msg.method && n.method && (n.method === "use")) {
                method = msg.method.toUpperCase(); // use the msg parameter
                // If the method is set to GET (by msg.method), we need to check what to do with the payload
                if (method === "GET") {
                    if (msg.payloadToQuerystring === true) { 
                        paytoqs = true
                    } else if (msg.payloadToBody === true) { 
                        paytobody = true
                    }
                }
            } else {
                  // use UI options (original code)
                if (n.paytoqs === true || n.paytoqs === "query") { 
                    paytoqs = true
                } else if (n.paytoqs === "body") { 
                    paytobody = true
                }
            }
```

Thoughts please Dave?

**Comment by marcus-j-davies** [comment]
Makes complete sense to include it in the message
if I may add a slight alterantive to consider?



```javascript
if (method === "GET") {
    switch (msg.payloadHandling.toLowerCase()) {
        case 'qs':
            paytoqs = true
            break
        case 'body':
            paytobody = true
            break
    }
} else {
  ...
}
```
Could make it easier to add more options in the future, using just one key to dictate handling.

**Comment by Steve-Mcl** [comment]
@marcus-j-davies I like the single msg property idea - it does simplify things somewhat. But for compatibility reasons, I would suggest the value options are `true`, `"query"` and `"body"`.  This aligns with the current option values.
* `true` and `"query"` would mean add payload to querystring
* `"body"` would mean add payload to body

The built in docs would not bother to mention the `true` value but for compatibility reasons, would be supported. 


```js
            let paytoqs = false
            let paytobody = false
            let payloadHandling = n.paytoqs
            if (msg.method && n.method && (n.method === "use")) {
                method = msg.method.toUpperCase(); // use the msg parameter
                // If the method is set to GET (by msg.method), use msg.payloadHandling
                payloadHandling = msg.payloadHandling
            }

            // Test payloadHandling to see if it has been set
            if (payloadHandling === true || payloadHandling === "query") { 
                paytoqs = true
            } else if (payloadHandling === "body") { 
                paytobody = true
            }
```

**Comment by dceejay** [comment]
Sounds sensible to me... no objections here.
